src/MSBuild/LiveLogger/LiveLogger.cs

```diff
@@ -134,16 +136,19 @@ private void eventSource_ProjectStarted(object sender, ProjectStartedEventArgs e
             // Get project id
             int id = e.BuildEventContext!.ProjectInstanceId;
 
-            // If id does not exist...
-            if (!projects.ContainsKey(id))
+            // Add or update project
+            ProjectNode node = new(e)
             {
-                // Add project
-                ProjectNode node = new(e)
-                {
-                    ShouldRerender = true,
-                };
-                projects[id] = node;
-                UpdateFooter();
+                ShouldRerender = true,
+            };
+            projects[id] = node;
+            UpdateFooter();
+
+            if (e.TargetNames?.Contains("Restore") == true)
+            {
+                TerminalBuffer.IsRestoring = true;
+                restoringProjects++;
```

Comment:
I don't understand this counter--I expected something more like "remember the ID of the first project that starts `Restore` and be done when that project instance completes". Does that not work in some circumstance?